### PR TITLE
Security: STG_PanelRelAuto_CTRL.cls doesnt seem to enforce user profile permissions or user field level security

### PR DESCRIPTION
**Pull request by dfuller114**
Added CRUD permission checks to the `Relationships Autocreation` page in NPSP Settings.
# Critical Changes

# Changes
- We improved security around the Relationship Autocreation NPSP Setting.
# Issues Closed

# Community Ideas Delivered

# New Metadata

# Deleted Metadata

### CODE REVIEW

**Comment by dfuller114** [comment]
**lurch: attach W-041515

**Comment by LurchTheButler** [comment]
Tracking <a href="https://foundation.my.salesforce.com/a2x1E000007haSiQAI" target="_blank">W-041515</a>

File: src/labels/CustomLabels.labels [hunk 1]
```diff
@@ -4921,6 +4921,14 @@ To check your GAU Allocation settings, go to Donations | GAU Allocations.</value
         <shortDescription>Percent</shortDescription>
         <value>Percent</value>
     </labels>
+    <labels>
+        <fullName>commonPermissionErrorMessage</fullName>
+        <categories>Gift Entry</categories>
+        <language>en_US</language>
+        <protected>true</protected>
+        <shortDescription>Page-level error for database level security check</shortDescription>
+        <value>You dont have the necessary access to use this feature. Please check with your Salesforce admin.</value>
```
Comment (melissabarber) [inline src/labels/CustomLabels.labels:4930]:
```suggestion
        <value>You don't have the necessary access to use this feature. Please check with your Salesforce Administrator.</value>
```

File: src/labels/CustomLabels.labels [hunk 2]
```diff
@@ -4921,6 +4921,14 @@ To check your GAU Allocation settings, go to Donations | GAU Allocations.</value
         <shortDescription>Percent</shortDescription>
         <value>Percent</value>
     </labels>
+    <labels>
+        <fullName>commonPermissionErrorMessage</fullName>
+        <categories>Gift Entry</categories>
+        <language>en_US</language>
+        <protected>true</protected>
+        <shortDescription>Page-level error for database level security check</shortDescription>
```
Comment (melissabarber) [inline src/labels/CustomLabels.labels:4929]:
```suggestion
        <shortDescription>Page-level error for database-level security check</shortDescription>
```

File: src/aura/STG_PanelRelAutoApp/STG_PanelRelAutoApp.app [hunk 3]
```diff
@@ -0,0 +1,5 @@
+<aura:application access="global" extends="ltng:outApp"
+                  description="Aura dependency application for STG_PanelRelAuto VF page">
+
+    <aura:dependency resource="c:utilIllustration" />
```
[the remark this answers is not in the document]
Comment (dfuller114) [reply src/aura/STG_PanelRelAutoApp/STG_PanelRelAutoApp.app:4]: @psadie-sfdo thanks for mentioning that. just pushed up that change.

File: src/classes/STG_PanelRelAuto_CTRL.cls [hunk 4]
```diff
@@ -37,6 +37,7 @@ public with sharing class STG_PanelRelAuto_CTRL extends STG_Panel {
 
     /*********************************************************************************************************
     * @description Returns the string Id of the Autocreate Relationships panel.
+     * @return String
```
[the remark this answers is not in the document]
Comment (dfuller114) [reply src/classes/STG_PanelRelAuto_CTRL.cls:40]: done!

File: src/classes/STG_PanelRelAuto_CTRL_TEST.cls [hunk 5]
```diff
@@ -0,0 +1,29 @@
+@IsTest
```
[the remark this answers is not in the document]
Comment (dfuller114) [reply src/classes/STG_PanelRelAuto_CTRL_TEST.cls:1]: done!

File: src/classes/STG_PanelRelAuto_CTRL.cls [hunk 6]
```diff
@@ -52,6 +53,15 @@ public with sharing class STG_PanelRelAuto_CTRL extends STG_Panel {
         private set;
     }
 
+    public Boolean getHasAccess() {
```
[the remark this answers is not in the document]
Comment (dfuller114) [reply src/classes/STG_PanelRelAuto_CTRL.cls:60]: done!

File: src/labels/CustomLabels.labels (hunk 2 above)
Comment (dfuller114) [reply src/labels/CustomLabels.labels:4929]: @melissabarber just pushed up that change.

**Comment by melissabarber** [comment]
Took a look at the updated labels. I'm good to go on this PR.

File: src/aura/RD2_EnablementApp/RD2_EnablementApp.app [hunk 7]
```diff
@@ -37,4 +37,5 @@
   <aura:dependency resource="c:RD2_EnablementDelegate" />
   <aura:dependency resource="c:RD2_EnablementDelegateSimple" />
   <aura:dependency resource="c:rd2StatusMappingSettings" />
+  <aura:dependency resource="c:utilIllustration" />
```
Comment (psadie-sfdo) [inline src/aura/RD2_EnablementApp/RD2_EnablementApp.app:40]: 👍

**Comment by dfuller114** [comment]
> Could we go ahead and clear out the `STG_PanelRelAutoApp` lightning app?

@psadie-sfdo Removed and pushed!

**Pull request merged**